torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -509,28 +510,26 @@ void ProcessGroupAgent::pollTimedOutRPCs() {
       return;
     }
 
-    const auto timedOutFutures = processTimedOutFutures();
+    const auto timedOutFutures = processTimedOutFutures(std::move(lock));
 
-    // Do not hold the lock while marking futures completed, as markCompleted()
-    // could invoke callbacks.
-    lock.unlock();
     for (const auto& timedOutFuture : timedOutFutures) {
       std::ostringstream ss;
       ss << "RPC ran for more than " << timedOutFuture.timeout_.count()
          << " milliseconds and timed out.";
       const auto exceptionMsg = createExceptionResponse(
           Message({}, {}, MessageType::EXCEPTION), ss.str());
+      // Do not hold the lock while marking futures completed, as
```

Comment:
Although I can see the argument for this comment being here, maybe it should stay above where the lock.unlock() was to explain that?

Reply:
Actually, this is not a valid comment. when using the future, we don't need to hold the lock. The lock is for protesting the future maps.